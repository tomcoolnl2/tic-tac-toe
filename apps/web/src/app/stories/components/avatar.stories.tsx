import type { StoryFn, Meta } from '@storybook/react';
import { PlayerSymbol } from '../../core/model';
import { Avatar, Props } from '../../components/avatar/avatar';


const Story: Meta<typeof Avatar> = {
    title: 'Components/Avatar',
    component: Avatar,
};
export default Story;

const Template: StoryFn<Props> = (args) => <Avatar {...args} />;

export const Default = Template.bind({});
Default.args = {
    type: PlayerSymbol.X,
    size: 'm',
};

export const LargeAvatar = Template.bind({});
LargeAvatar.args = {
    type: PlayerSymbol.O,
    size: 'xl',
};

export const DarkVariant = Template.bind({});
DarkVariant.args = {
    type: PlayerSymbol.X,
    size: 's',
    variant: 'dark',
};

export const LightVariant = Template.bind({});
LightVariant.args = {
    type: PlayerSymbol.O,
    size: 'l',
    variant: 'light',
};