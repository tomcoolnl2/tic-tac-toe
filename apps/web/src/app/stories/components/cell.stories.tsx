import { StoryFn, type Meta } from '@storybook/react';
import { PlayerSymbol } from '../../core/model';
import { Cell, Props } from '../../components/cell/cell';


const Story: Meta<typeof Cell> = {
    component: Cell,
    title: 'Components/Cell',
};
export default Story;


const Template: StoryFn<Props> = (args) => <Cell {...args} />;

export const CellX = Template.bind({});
CellX.args = {
    type: PlayerSymbol.X,
    index: 1,
    solutionCells: [0, 0, 0],
};

export const CellO = Template.bind({});
CellO.args = {
    type: PlayerSymbol.O,
    index: 2,
    solutionCells: [0, 0, 0],
};

export const InvertedCellX = Template.bind({});
InvertedCellX.args = {
    type: PlayerSymbol.X,
    index: 2,
    solutionCells: [0, 1, 2],
};

export const InvertedCellO = Template.bind({});
InvertedCellO.args = {
    type: PlayerSymbol.O,
    index: 2,
    solutionCells: [0, 1, 2],
};