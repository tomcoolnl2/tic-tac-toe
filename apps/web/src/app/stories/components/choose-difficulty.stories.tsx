import { StoryFn, type Meta } from '@storybook/react';
import { IntelligenceLevel } from '../../core/model';
import { ChooseDifficulty, Props } from '../../components/choose-difficulty/choose-difficulty';
import { withContainer, type ContainerProps } from '../helpers';


const EnchantedChooseDifficulty: React.FC<Props & ContainerProps> = withContainer(ChooseDifficulty, 320);
const Story: Meta<typeof ChooseDifficulty> = {
    component: ChooseDifficulty,
    title: 'Components/ChooseDifficulty',
};
export default Story;


const Template: StoryFn<Props> = (args) => <EnchantedChooseDifficulty {...args} />;

export const ChooseDifficultyEasy = Template.bind({});
ChooseDifficultyEasy.args = {
    selectedDifficultySetting: IntelligenceLevel.BIEBER
};

export const ChooseDifficultyNovice = Template.bind({});
ChooseDifficultyNovice.args = {
    selectedDifficultySetting: IntelligenceLevel.NOVICE
};

export const ChooseDifficultyMaster = Template.bind({});
ChooseDifficultyMaster.args = {
    selectedDifficultySetting: IntelligenceLevel.MASTER
};