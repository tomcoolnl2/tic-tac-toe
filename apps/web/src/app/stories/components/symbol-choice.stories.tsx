import { StoryFn, type Meta } from '@storybook/react';
import { SymbolChoice, Props } from '../../components/symbol-choice/symbol-choice';
import { PlayerSymbol } from '../../core/model';
import { withContainer, type ContainerProps } from '../helpers';

const EnchantedSymbolChoice: React.FC<Props & ContainerProps> = withContainer(SymbolChoice, 320);
const Story: Meta<typeof SymbolChoice> = {
    component: EnchantedSymbolChoice,
    title: 'Components/SymbolChoice',
};
export default Story;

  
const Template: StoryFn<Props> = (args) => <EnchantedSymbolChoice {...args} />;

export const ChoosePlayerO = Template.bind({});
ChoosePlayerO.args = {
    playerSymbol: PlayerSymbol.O,
    handleSymbolChoiceChange: (): null => null
};

export const ChoosePlayerX = Template.bind({});
ChoosePlayerX.args = {
    playerSymbol: PlayerSymbol.X,
    handleSymbolChoiceChange: (): null => null
};
  