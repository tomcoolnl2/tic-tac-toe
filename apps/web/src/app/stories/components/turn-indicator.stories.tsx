import { StoryFn, type Meta } from '@storybook/react';
import { TurnIndicator, Props } from '../../components/turn-indicator/turn-indicator';
import { PlayerSymbol } from '../../core/model';
import { withContainer, type ContainerProps } from '../helpers';


const EnchantedTurnIndicator: React.FC<Props & ContainerProps> = withContainer(TurnIndicator, 90);
const Story: Meta<typeof TurnIndicator> = {
    title: 'Components/TurnIndicator',
    component: TurnIndicator,
};
export default Story;


const Template: StoryFn<Props> = (args) => <EnchantedTurnIndicator {...args} />;

export const TurnX = Template.bind({});
TurnX.args = {
    currentPlayer: PlayerSymbol.X,
};
  
export const TurnO = Template.bind({});
TurnO.args = {
    currentPlayer: PlayerSymbol.O,
};
