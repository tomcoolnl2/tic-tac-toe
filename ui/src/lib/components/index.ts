export { AppLogo } from './logo/logo';
export { Avatar } from './avatar/avatar';
export { Button } from './button/button';
export { Cell } from './cell/cell';
export { ChooseDifficulty } from './choose-difficulty/choose-difficulty';
export { GameOver } from './game-over/game-over';
export { Input } from './input/input';
export { Modal } from './modal/modal';
export { ReloadButton } from './reload-button/reload-button';
export { ScoreBoardItem } from './score-board-item/score-board-item';
export { Spinner } from './spinner/spinner';
export { SymbolChoice } from './symbol-choice/symbol-choice';
export { TurnIndicator } from './turn-indicator/turn-indicator';
