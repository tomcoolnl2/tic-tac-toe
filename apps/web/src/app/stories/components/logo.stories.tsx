import type { Meta } from '@storybook/react';
import { AppLogo } from '../../components';

const Story: Meta<typeof AppLogo> = {
    component: AppLogo,
    title: 'Components/AppLogo',
};
export default Story;

export const Default = {
    args: {},
};
