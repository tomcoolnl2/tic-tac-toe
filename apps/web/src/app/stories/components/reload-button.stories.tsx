import type { Meta } from '@storybook/react';
import { ReloadButton } from '../../components';


const Story: Meta<typeof ReloadButton> = {
    component: ReloadButton,
    title: 'Components/ReloadButton',
};
export default Story;

export const Default = {
    args: { 
        disabled: false,
    },
};
