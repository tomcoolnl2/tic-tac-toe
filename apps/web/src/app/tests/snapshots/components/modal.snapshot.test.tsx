
import { render } from '@testing-library/react'
import { Modal } from '../../../components'

test('Modal component snapshot test', () => {
    const { container } = render(
        <Modal>
            <div />
        </Modal>
    )
    expect(container).toMatchSnapshot()
})
