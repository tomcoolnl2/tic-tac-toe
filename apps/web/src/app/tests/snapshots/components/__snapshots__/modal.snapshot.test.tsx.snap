// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component snapshot test 1`] = `
<div>
  <div
    class="flex-box direction-row justify-content-flex-start align-items-center modal-backdrop"
  >
    <div
      class="flex-box direction-column justify-content-flex-start align-items-center modal-background"
    >
      <div
        class="flex-box direction-column justify-content-center align-items-center modal-content"
      >
        <div />
      </div>
    </div>
  </div>
</div>
`;
